js/ui/grid_core/ui.grid_core.validating.js

```diff
@@ -403,20 +414,40 @@ module.exports = {
                 },
 
                 _afterInsertRow: function(options) {
+                    this._createInvisibleColumnValidators([options]);
+
+                    this.callBase(options);
+                },
+
+                _createInvisibleColumnValidators: function(options) {
                     var validatingController = this.getController("validating"),
-                        invisibleColumns = commonUtils.grep(this.getController("columns").getInvisibleColumns(), function(column) { return !column.isBand; });
+                        invisibleColumns = commonUtils.grep(this.getController("columns").getInvisibleColumns(), function(column) { return !column.isBand; }),
+                        invisibleColumnValidators = [];
 
                     if(FORM_BASED_MODES.indexOf(this.getEditMode()) === -1) {
                         each(invisibleColumns, function(_, column) {
-                            validatingController.createValidator({
-                                column: column,
-                                key: options.key,
-                                value: column.calculateCellValue(options.data)
+                            options.forEach(function(option) {
+                                if(option.type === "insert" || option.data && option.data.hasOwnProperty(column.dataField) && option.data[column.dataField] !== null) {
+                                    invisibleColumnValidators.push(validatingController.createValidator({
+                                        column: column,
+                                        key: option.key,
+                                        value: option.data[column.dataField]
```

Comment:
Changing from column.calculateCellValue(options.data) to option.data[column.dataField] is wrong. For example, complex fields will be broken.